### ISSUE TITLE
Exception Call to a member function quote() on null when saving records

### ISSUE THREAD

**Issue by marlongichie** [barryvdh/laravel-debugbar#306]
The debug toolbar works as expected when loading pages and browsing.

However when I try to save a record to the database, I get a:

```
'Symfony\Component\Debug\Exception\FatalErrorException' with message 'Call to a member function quote() on null'
```

The error suggests it is coming from: 

```
vendor/barryvdh/laravel-debugbar/src/DataCollector/QueryCollector.php:102
```

When I disable the toolbar with config.php, records are saved as expected with no errors.

Anyone else experiencing this issue? Any suggestions?

**Comment by tomzx** [comment]
Which version of the debugbar are you using?

**Comment by marlongichie** [comment]
Hello @tomzx , thanks for getting back.

My composer require is `"barryvdh/laravel-debugbar": "~1.8",`.

The issue it seems, is that I am using the `neo4j` database via the [NeoEloquent Package] (https://github.com/Vinelab/NeoEloquent) which does not have a `pdo` instance (returns null) and hence throws the `'Call to a member function quote() on null'` error.

I was able to work around it by hard coding it to use the `mysql` pdo instance like so 
```$pdo = DB::connection('mysql')->getPdo();```

That seemed to work fine, but it's by no means a permanent solution.

Your thoughts?

**Comment by tomzx** [comment]
From the little I know about NeoEloquent, my guess would be that it uses Laravel `DatabaseManager` to manage its connection. `DB::connection()` is said to return a `\Illuminate\Database\Connection` instance, which has the `getPdo()` method.

I think that Laravel's `DatabaseManager` implementation expects only to manage PDO connections, nothing else. The fact that NeoEloquent extends `\Illuminate\Database\Connection` in its own `Vinelab\NeoEloquent\Connection` also points that there's something wrong. At best, his class should implement `Illuminate\Database\ConnectionInterface`.

*tl;dr* Laravel-debugbar is not meant to support NeoEloquent. Either it should check if getPdo() returns something and if so call `->quote()` or use a generic `quote` method OR ignore/drop the query altogether (so no query logging for custom `Connection`).
